specs/directory.md

```diff
@@ -0,0 +1,387 @@
+| `CONFIG`  | `/etc`         | `%ProgramData%`               | `/Library/Application Support` |
+
+`CACHE` is omitted since it is user specific.
+
+### User Level
+
+Default directory paths for various operating systems at user level are specified as below.
+
+| Directory | Unix               | Windows          | Darwin                          |
+| --------- | ------------------ | ---------------- | ------------------------------- |
+| `LIBEXEC` | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CONFIG`  | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CACHE`   | `$XDG_CACHE_HOME`  | `%LocalAppData%` | `~/Library/Caches`              |
+
+There is no default `BIN` path at user level since the `notation` binary can be put anywhere as long as it in the `PATH` environment variable. Common directories on Unix/Darwin are `~/bin` and `~/.local/bin` where manual `PATH` update by users may be required.
+
+## Structure
+
+The overall directory structure for `notation` is summarized as follows.
+
+```
+{BIN}
+└── notation
+{CACHE}
+└── notation
+    └── signature
+        └── {manifest-digest-algorithm}
+            └── {manifest-digest}
+                └── {signature-digest-algorithm}
+                    └── {signature-digest}.sig
+{CONFIG}
+└── notation
+    ├── config.json
+    ├── signing
+    │   ├── certificates
+    │   │   └── {key-name}.crt
+    │   ├── keys
+    │   │   └── {key-name}.pem
+    │   └── keys.json
+    └── trust
+        ├── policy.json
+        └── store
+            └── {trust-store-type}
+                └── {named-store}
+                    └── {cert-file}
+{LIBEXEC}
+└── notation
+    └── plugins
+        └── {plugin-name}
+            └── notation-{plugin-name}
+```
+
+### Notation Binary
+
+The path for the `notation` binary is
+
+```
+{BIN}/notation
+```
+
+On Windows, the `.exe` extension is required for executables.
+
+```
+{BIN}/notation.exe
+```
+
+### Plugin
+
+[Plugins][Plugin] are binaries not meant to be executed directly by users' shell or scripts. The path of a plugin follows the pattern below.
+
+```
+{LIBEXEC}/notation/plugins/{plugin-name}/notation-{plugin-name}
+```
+
+On Windows, the `.exe` extension is required for executables.
+
+```
+{LIBEXEC}/notation/plugins/{plugin-name}/notation-{plugin-name}.exe
+```
+
+### General Configuration
+
+The path of the general configuration file of the `notation` CLI is
+
+```
+{CONFIG}/notation/config.json
+```
+
+### Trust Store
+
+The path of a certificate file in a [Trust Store][TS] follows the pattern of
+
+```
+{CONFIG}/notation/trust/store/{trust-store-type}/{named-store}/{cert-file}
+```
+
+### Trust Policy
+
+The path of the [Trust Policy][TP] file is
+
+```
+{CONFIG}/notation/trust/policy.json
+```
+
+### Signature Caches
+
+The signatures are cached to optimize the network traffic. The path of a cached signature for a certain manifest follows the pattern below.

```

Comment:
Can you provide some more details here for my benefit, we can probably capture the details in another doc. How does caching work, given that additional signatures will be added over time. Will notation-go always call references API and skip downloading signature blobs it has already downloaded? We still have to download signature manifests every time, correct?